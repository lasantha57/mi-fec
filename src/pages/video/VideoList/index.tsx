import React, { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { Typography, Box, TextField } from '@mui/material';
import { Container } from '@mui/system';

import { ProcessedVideo } from '../../../common/interfaces';
import { deleteVideo, getVideos } from '../../../services/videos';
import { VideosTable } from '../../../components/VideosTable';

export const VideoList: React.FC = () => {
  const navigate = useNavigate();
  const [videos, setVideos] = useState<ProcessedVideo[]>([]);
  const [searchText, setSearchText] = useState<string>('');

  const fetchVideos = async () => {
    try {
      const videos = await getVideos();
      setVideos(videos);
    } catch (error) {
      console.log(error);
    }
  }

  useEffect(() => {
    fetchVideos();
  }, []);

  const onDelete = async (videoId: number, authorId: number) => {
    try {
      const deleted = await deleteVideo(authorId, videoId);
      if (deleted) {
        fetchVideos();
      }
    } catch (error) {
      console.log(error)
    }
  }

  const onEdit = (videoId: number, authorId: number) => {
    navigate({
      pathname: '/video',
      search: `?authorId=${authorId}&videoId=${videoId}`,
    });
  }

  return (
    <Container maxWidth={false}>
      <Box>
        <Typography variant="h4" my={2}>
          VManager Demo v0.0.1
        </Typography>
        <TextField
          value={searchText}
          onChange={(event) => setSearchText(event.target.value)}
          label="Search Videos"
          type={'search'}
          fullWidth
        />
        <Box my={3}>
          <VideosTable
            videos={videos}
            searchText={searchText}
            onDelete={onDelete}
            onEdit={onEdit}
          />
        </Box>
      </Box>
    </Container>
  );
};
