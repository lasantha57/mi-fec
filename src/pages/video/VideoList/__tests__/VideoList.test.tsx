import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';

import VideoList from '..';

afterEach(cleanup)

describe('Testing VideoList', () => {
    it('Component should render correctly', () => {
        const tree = renderer
            .create(
                <Router>
                    <VideoList />
                </Router>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});