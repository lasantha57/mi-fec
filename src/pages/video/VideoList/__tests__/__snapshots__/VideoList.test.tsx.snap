// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing VideoList Component should render correctly 1`] = `
<div
  className="MuiContainer-root css-ur2jdm-MuiContainer-root"
>
  <div
    className="MuiBox-root css-0"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 css-ue75iy-MuiTypography-root"
    >
      VManager Demo v0.0.1
    </h4>
    <div
      aria-label="Search Videos"
      className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor=":r0:"
        id=":r0:-label"
      >
        Search Videos
      </label>
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputTypeSearch css-1g24dm6-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r0:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="search"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-1ftyaf0"
          >
            <span>
              Search Videos
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiBox-root css-5vb4lz"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-1xg4c7-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          aria-label="A Video table"
          className="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
          role={null}
        >
          <thead
            className="MuiTableHead-root css-15bsjcf-MuiTableHead-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
              >
                Video Name
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
              >
                Author
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
              >
                Categories
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
              >
                Highest Quality Format
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
              >
                Release Date
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                scope="col"
              >
                Options
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            role={null}
          />
          <tfoot
            className="MuiTableFooter-root css-9cgc89-MuiTableFooter-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root MuiTableRow-footer css-1q1u3t4-MuiTableRow-root"
              role={null}
            >
              <td
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-footer MuiTableCell-sizeMedium MuiTablePagination-root css-fikjyc-MuiTableCell-root-MuiTablePagination-root"
                colSpan={1000}
              >
                <div
                  className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
                >
                  <div
                    className="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
                  />
                  <p
                    className="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
                    id=":r2:"
                  >
                    Rows per page:
                  </p>
                  <div
                    className="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
                    onClick={[Function]}
                    variant="standard"
                  >
                    <div
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-labelledby=":r2: :r1:"
                      className="MuiTablePagination-select MuiSelect-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
                      id=":r1:"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onMouseDown={[Function]}
                      role="button"
                      tabIndex={0}
                    >
                      5
                    </div>
                    <input
                      aria-hidden={true}
                      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                      onAnimationStart={[Function]}
                      onChange={[Function]}
                      tabIndex={-1}
                      value={5}
                    />
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTablePagination-selectIcon MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </div>
                  <p
                    className="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
                  >
                    0â€“0 of 0
                  </p>
                  <div
                    className="MuiTablePagination-actions"
                  >
                    <button
                      aria-label="Go to previous page"
                      className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                      disabled={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={-1}
                      title="Go to previous page"
                      type="button"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="KeyboardArrowLeftIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                        />
                      </svg>
                    </button>
                    <button
                      aria-label="Go to next page"
                      className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                      disabled={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={-1}
                      title="Go to next page"
                      type="button"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="KeyboardArrowRightIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </td>
            </tr>
          </tfoot>
        </table>
      </div>
    </div>
  </div>
</div>
`;
